python/src/com/jetbrains/python/psi/types/PyTypeChecker.java

```diff
@@ -393,11 +393,26 @@ else if (type instanceof PyCallableType) {
     final Map<PyGenericType, PyType> substitutions = unifyReceiver(receiver, context);
     for (Map.Entry<PyExpression, PyNamedParameter> entry : arguments.entrySet()) {
       final PyNamedParameter p = entry.getValue();
-      if (p.isPositionalContainer() || p.isKeywordContainer()) {
-        continue;
-      }
-      final PyType argType = context.getType(entry.getKey());
       final PyType paramType = context.getType(p);
+
+      PyType argType = context.getType(entry.getKey());
+      if (p.isPositionalContainer()) {
+        final PyClassType tupleClassType = PyBuiltinCache.getInstance(p).getTupleType();
+        if (tupleClassType == null) {
+          return null;
+        }
+        final PyClass tupleClass = tupleClassType.getPyClass();
+        argType = new PyCollectionTypeImpl(tupleClass, false, argType);
```

Comment:
We now have `new PyTupleType(tupleClass, elementTypes, true)` for homogeneous tuples in our type system.